src/lib/generate-files.ts

```diff
@@ -1,64 +1,81 @@
-import * as path from 'path';
-import * as fs from 'fs';
+import * as path from "path";
+import * as fs from "fs";
 import * as prettier from "prettier";
-import debug from 'debug';
-import chalk from 'chalk';
-import { confirm } from '@inquirer/prompts';
+import debug from "debug";
+import chalk from "chalk";
+import { confirm } from "@inquirer/prompts";
 
 const SUBSTITUTION_PATTERN = /__([^_]+)__/g;
-export const TEMPLATE_EXTENSION = '.ejs';
+export const TEMPLATE_EXTENSION = ".ejs";
 
-debug.enable('generator');
-const log = debug('generator');
+debug.enable("generator");
+const log = debug("generator");
 
 export enum OverwriteStrategy {
-  Overwrite = 'overwrite',
-  KeepExisting = 'keepExisting',
-  ThrowIfExisting = 'throwIfExisting',
-  Prompt = 'prompt',
+  Overwrite = "overwrite",
+  KeepExisting = "keepExisting",
+  ThrowIfExisting = "throwIfExisting",
+  Prompt = "prompt",
 }
 
+export type ParserOptions =
+  | "typescript"
+  | "javascript"
+  | "json"
+  | "css"
+  | "vue";
 export interface GenerateFilesOptions {
   overwriteStrategy?: OverwriteStrategy;
-  parser?: 'typescript' | 'javascript' | 'json' | 'css';
+  parser?: ParserOptions | ((filename: string) => ParserOptions);
```

Comment:
This is one change, making it possible to pass a `parser function` to distinguish different files in the same `generateFiles` batch by their own extensions